packages/core/lib/modules/account.js

```diff
@@ -0,0 +1,94 @@
+import {Accounts} from './accounts';
+
+export type AccountDetails = {
+  id: string,
+  name: string,
+  type: KeypairType,
+  address: string,
+  correlations: any[],
+};
+
+export class Account {
+  details: AccountDetails;
+  address: string;
+  name: string;
+  accounts: Accounts;
+
+  constructor(address: string, accounts: Accounts) {
+    this.accounts = accounts;
+    this.address = address;
+    this.id = address;
+  }
+
+  static with(address) {
+    const account = new Account(address, Accounts.getInstance());
+
+    account.loadPromise = account.loadDetails();
+
+    return account;
+  }
+
+  static async withAsync(address) {
+    const account = Account.with(address);
+
+    await account.loadPromise;
+
+    return account;
+  }
+
+  async loadDetails() {
+    this.details = await this.accounts.wallet.getDocumentById(this.address);
+    // this.name = this.details.name;
+  }
+
+  getAddress() {
+    return this.details.address;
```

Comment:
what if this is called before `loadDetails()`? Right now you'll get an undefined exception - could we throw a more useful error instead? e.g. `you must all loadDetails() first` or something?

Reply:
this applies for pretty well all the functions below